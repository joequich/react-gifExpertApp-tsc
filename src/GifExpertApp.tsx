import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

interface Props {
    defaultCategories?: string[];
}

export const GifExpertApp = ( {defaultCategories = []}: Props) => {
    const [ categories, setCategories ] = useState<string[]>(defaultCategories);

    return (
        <>
            <h2 className="title">
                <span className="title-text">GifExpertApp</span>
            </h2>
            <AddCategory setCategories={setCategories} />
            <hr/>
            <ol>
                {categories.map( category => (
                    <GifGrid 
                        key={category}
                        category={category} 
                    />
                ))}
            </ol>
        </>
    );
}