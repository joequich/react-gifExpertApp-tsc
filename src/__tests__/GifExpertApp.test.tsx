import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../GifExpertApp';

describe('GifExpertApp', () => {
    it('render list of categories', () => {
        const categories = ['music', 'movies'];
        const { asFragment } = render(<GifExpertApp defaultCategories={categories}/>);
        // expect(asFragment()).toMatchSnapshot();
        expect(screen.queryAllByRole('list')).toMatchSnapshot();
        // expect(screen.queryAllByRole('list').length).toBe(categories.length);
    });
});