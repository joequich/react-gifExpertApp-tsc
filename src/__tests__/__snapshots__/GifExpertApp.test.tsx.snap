// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp render list of categories 1`] = `
Array [
  <ol>
    <li>
      <h3
        class="category-title"
      >
        music
      </h3>
      <p
        class="animate__pulse"
      >
        Loading...
      </p>
      <div
        class="card-grid animate__slideInUp"
        role="list"
      />
    </li>
    <li>
      <h3
        class="category-title"
      >
        movies
      </h3>
      <p
        class="animate__pulse"
      >
        Loading...
      </p>
      <div
        class="card-grid animate__slideInUp"
        role="list"
      />
    </li>
  </ol>,
  <div
    class="card-grid animate__slideInUp"
    role="list"
  />,
  <div
    class="card-grid animate__slideInUp"
    role="list"
  />,
]
`;
